scaladoc/src/dotty/tools/scaladoc/renderers/HtmlRenderer.scala

```diff
@@ -58,19 +58,18 @@ class HtmlRenderer(rootPackage: Member, val members: Map[DRI, Member])(using ctx
 
   val hiddenPages: Seq[Page] =
     staticSite match
-      case None =>
-        Seq(navigablePage.copy( // Add index page that is a copy of api/index.html
-          link = navigablePage.link.copy(dri = docsRootDRI),
-          children = Nil
-        ))
+      case None => Nil
       case Some(siteContext) =>
-        (siteContext.orphanedTemplates :+ siteContext.indexTemplate()).map(templateToPage(_, siteContext))
+        val actualIndexTemplate = siteContext.indexTemplates() match
+          case Nil if effectiveMembers.isEmpty => Seq(siteContext.emptyIndexTemplate)
+          case templates => templates
+
+        (siteContext.orphanedTemplates ++ actualIndexTemplate).map(templateToPage(_, siteContext))
```

Comment:
Could you add a comment explaining what this does? I’m a bit lost.

Reply:
It seems it just a rewrite (no new logic is added here). 

Before:
we got one file by calling `siteContext.indexTemplate()`

Now:
we call `siteContext.indexTemplates()` which returns collection (effectively only one file), then we pattern match on it just to replace it with `emptyIndextemplate` (which was embedded in old `siteContext.indexTemplate()`) if it was empty or get `identity` otherwise and then append it as a collection.

Can you @romanowski explain if I get something wrong here? The only thing is granularity (we have more designated functions to call), but on the other hand main misconception here is that `indexTemplates()` actually can yield 0 or 1 file, never 2.